### PR TITLE
Minimal update to make utop build on OCaml 4.07.0+rc1

### PR DESCRIPTION
**Pull request by hcarty**
This is not the _right_ fix.  It just allows `make` to complete.

### CODE REVIEW

File: src/lib/uTop_complete.ml [hunk 1]
```diff
@@ -541,6 +541,25 @@ let list_global_names () =
     | Env.Env_constraints (summary, _) ->
         loop acc summary
 #endif
+#if OCAML_VERSION >= (4, 07, 0)
+    | Env.Env_copy_types (summary, _) ->
+        loop acc summary
+#endif
+#if OCAML_VERSION >= (4, 07, 0)
```
Comment (diml) [inline src/lib/uTop_complete.ml]: The conditions in these two blocks is the same, shouldn't they be conflated?

**Comment by diml** [comment]
Why is not the right fix BTW?

**Comment by hcarty** [comment]
@diml I didn't do much looking into the new/changed constructors so I'm not sure if there are further changes to `uTop_complete.ml` required to handle the constructors properly.  I'm not familiar with utop's internal or [this comment](https://github.com/ocaml/ocaml/blob/4197e75b6b864c787cdf6162c1303dd74e1e3d5b/typing/env.mli#L33-L34) upstream so I don't want to presume that this minimal update is correct. 

I'll simplify the `#if`ing as you suggested [here](https://github.com/diml/utop/pull/238#discussion_r197054574).  If you're happy with the PR at that point I'm all for merging so utop can be ready for 4.07.0!

**Comment by hcarty** [comment]
@diml Simplified - the overall diff is fairly minimal now.

**Comment by diml** [comment]
> I'm not familiar with utop's internal or this comment upstream so I don't want to presume that this minimal update is correct.

Ok. This comment was actually from one of my PR :) I checked utop's code and convinced myself that there is nothing more to do here.

Thanks!

**Pull request merged**